src/System.Management.Automation/security/SecuritySupport.cs

```diff
@@ -1336,39 +1336,17 @@ public enum ResolutionPurpose
 
     internal static class AmsiUtils
     {
-        private static string GetProcessHostName(string processName)
-        {
-            return string.Concat("PowerShell_", processName, ".exe_0.0.0.0");
-        }
-
         internal static int Init()
         {
             Diagnostics.Assert(s_amsiContext == IntPtr.Zero, "Init should be called just once");
 
             lock (s_amsiLockObject)
             {
-                Process currentProcess = Process.GetCurrentProcess();
-                string hostname;
-                try
-                {
-                    hostname = string.Concat("PowerShell_", Environment.ProcessPath, "_",
-                        currentProcess.MainModule.FileVersionInfo.ProductVersion);
-                }
-                catch (ComponentModel.Win32Exception)
-                {
-                    // This exception can be thrown during thread impersonation (Access Denied for process module access).
-                    hostname = GetProcessHostName(currentProcess.ProcessName);
-                }
-                catch (FileNotFoundException)
-                {
-                    // This exception can occur if the file is renamed or moved to some other folder
-                    // (This has occurred during Exchange set up).
-                    hostname = GetProcessHostName(currentProcess.ProcessName);
-                }
+                string appName = string.Concat("PowerShell_", Environment.ProcessPath, "_", PSVersionInfo.ProductVersion);
```

Comment:
Can 'Environment.ProcessPath' throw?  We should have a try/catch here so that AMSI is always initialized.

Reply:
Actually yes, it might throw (https://github.com/dotnet/runtime/blob/a5e626a015a37cf332e993804cd8bb42943c14c2/src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs#L104).
Should I then catch a generic `Exception` here?

Reply:
Yes, I think a generic Exception, and then create the default hostname as before.

Reply:
I've added a catch block as you requested. I kept using `PSVersionInfo.ProductVersion` in the catch block (before the PR a placeholder version of `0.0.0.0` was used) since we can safely assume that it will never throw anything.

Reply:
Btw, I'm a bit uncertain on using Process.ProcessName as a "safe" fallback - it doesn't look much safer than Environment.ProcessPath given that it could potentially throw an InvalidOperationException (https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.processname?view=net-5.0#exceptions).

Reply:
Good point, but since it is basically the same as before at least we don't introduce a possible regression.

Reply:
If the assumption is that `Environment.ProcessPath` may throw while `PSVersionInfo.ProductVersion` will never throw, then a comment should be added in the catch block to call this out.